File: scripts/transform_server.py
#!/usr/bin/env python

PACKAGE = 'transforms'
import roslib;roslib.load_manifest(PACKAGE)
import rospy

from dynamic_reconfigure.server import Server
from transforms.cfg import TransformConfig
import tf
import tfx

class Broadcaster:
    
    def __init__(self, br):
        self.config = None
        self.srv = Server(TransformConfig, self.callback)
        self.br = br

    def callback(self, config, level):
        self.config = config
        return config

    def broadcast(self):
        if not self.config:
            return
        config = self.config
        self.br.sendTransform((config.x, config.y, config.z),
                         tfx.tb_angles(config.yaw, config.pitch, config.roll).quaternion,
                         rospy.Time.now(),
                         '/left_optical_frame',
                         '/world')

if __name__ == "__main__":
    rospy.init_node("transform_server")
    br = tf.TransformBroadcaster()

    bc = Broadcaster(br)
    r = rospy.Rate(100)
    while not rospy.is_shutdown():
        bc.broadcast()
        r.sleep()

File: src/transform_server.cpp
#include <ros/ros.h>

#include <dynamic_reconfigure/server.h>
#include <transforms/TransformConfig.h>

void callback(transforms::TransformConfig &config, uint32_t level) {
    ROS_INFO("x: %f", config.x);
}

int main(int argc, char **argv) {
  ros::init(argc, argv, "transforms");

  dynamic_reconfigure::Server<transforms::TransformConfig> server;
  dynamic_reconfigure::Server<transforms::TransformConfig>::CallbackType f;

  f = boost::bind(&callback, _1, _2);
  server.setCallback(f);

  ROS_INFO("Spinning node");
  ros::spin();
  return 0;
}
